packages/nx/src/tasks-runner/cache.ts

```diff
@@ -207,6 +225,32 @@ export class Cache {
       try {
         code = Number(await readFile(join(td, 'code'), 'utf-8'));
       } catch {}
+
+      let sourceMachineId = null;
+      try {
+        sourceMachineId = await readFile(join(td, 'source'), 'utf-8');
+      } catch {}
+
+      if (
+        sourceMachineId &&
+        sourceMachineId != (await this.currentMachineId())
+      ) {
+        if (
+          process.env.NX_REJECT_UNKNOWN_LOCAL_CACHE != '0' &&
+          process.env.NX_REJECT_UNKNOWN_LOCAL_CACHE != 'false'
```

Comment:
I thought we were going to warn even when it is set. Did you change your mind?

Reply:
Yeah I changed it for now. I can always add it later.